Simplenote/src/main/java/com/automattic/simplenote/utils/AccountVerificationWatcher.java

```diff
@@ -29,33 +27,44 @@
     }
 
     private final Simplenote simplenote;
-    private final List<VerificationStateListener> listeners = new ArrayList<>();
+    private final VerificationStateListener listener;
+    private Status currentState = Status.UNKNOWN;
 
-    public AccountVerificationWatcher(Simplenote simplenote) {
+    public AccountVerificationWatcher(Simplenote simplenote, VerificationStateListener listener) {
         this.simplenote = simplenote;
+        this.listener = listener;
     }
 
-    public void addVerificationStateListener(VerificationStateListener listener) {
-        listeners.add(listener);
+    private void notifyListener(Status status) {
+        if (listener != null) {
+            listener.onUpdate(status);
+        }
```

Comment:
since we set this in the constructor do we need the `null` check?

if we do, would it help to add `@NonNull VerificationStateListener listener` as the constructor arg? We use that around the app already.

Reply:
Yeah, let me add that annotation to remove the null check.

Reply:
Fixed in 5d421ed.